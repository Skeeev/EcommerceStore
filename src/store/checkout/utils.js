import get from 'lodash/get';
import find from 'lodash/find';

const normalizeProducts = products =>
  products.map(({ key, name: title, price, total, quantity, thumb }) => ({
    id: `${key}`,
    title,
    price,
    total,
    quantity,
    imageUri: `http:${thumb}`
  }));

export const extractCheckoutData = ({ products, totals }) => ({
  products: normalizeProducts(products),
  total: get(find(totals, { id: 'total' }), 'text', '$00.00'),
  totalQuantity: products.reduce((total, { quantity }) => total + quantity, 0)
});
