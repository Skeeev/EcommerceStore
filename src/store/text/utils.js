export const prepareTextPayload = text => ({
  payload: {
    text
  }
});
