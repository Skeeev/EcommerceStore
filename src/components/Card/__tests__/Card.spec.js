import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';

import { getComponentDisplayName } from 'src/utils/displayName';
import { Card } from '..';

describe(`${getComponentDisplayName(Card)} component`, () => {
  const renderComponent = props => shallow(<Card {...props} />);

  it('renders correctly', () => {
    const component = renderComponent({
      children: <View />
    });

    expect(component).toMatchSnapshot();
  });
});
