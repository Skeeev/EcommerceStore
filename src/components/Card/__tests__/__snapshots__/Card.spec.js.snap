// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 5,
      "elevation": 3,
      "flexDirection": "row",
      "marginBottom": 10,
      "marginHorizontal": 1,
      "marginTop": 1,
      "paddingHorizontal": 30,
      "paddingVertical": 15,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.22,
      "shadowRadius": 2.22,
    }
  }
>
  <View />
</TouchableOpacity>
`;
